tests/test_selection/test_base_selection_functions.py

```diff
@@ -184,6 +199,37 @@ def test_single_feature_performance_cv_generator(df_test):
         assert mean_ == expected_mean
 
 
+def test_single_feature_performance_with_groups(df_test_with_groups):
+    X, y, groups = df_test_with_groups
+    rf = RandomForestClassifier(n_estimators=5, random_state=1)
+    variables = X.columns.to_list()
+    scoring = "neg_mean_absolute_error"
+    cv = GroupKFold(n_splits=3)
+    cv_indices = cv.split(X=X, y=y, groups=groups)
+
+    expected_mean_, expected_std_ = single_feature_performance(
+        X=X,
+        y=y,
+        variables=variables,
+        estimator=rf,
+        cv=cv_indices,
+        scoring=scoring,
+    )
+
+    mean_, std_ = single_feature_performance(
+        X=X,
+        y=y,
+        variables=variables,
+        estimator=rf,
+        cv=cv,
+        scoring=scoring,
+        groups=groups,
```

Comment:
If I understand this test correctly, we are comparing the results of the function `single_feature_performance` when groups is `None` with the results returned when groups is the `group` variable, and the results are exactly the same, which comes from the dataset containing random values, I guess. But then, we can't be sure whether the grouping variable is working or not, if passing None or something else return the same thing.

As painful as it is, we need to add hard coded values, or return the expected values by an alternative function(s), that is independent of the function we are trying to test.

Reply:
if we have to make sure about the independent expected results for tests, then I will update it, otherwise let me explain the reason for the test logic I chose. Using Splitter object as cv and giving groups is clear, and for expected results, I do not use groups, and manually find the train and validation indexes for splits considering existence of the groups, then pass generator for list of indices to the class, so this is expected result that current feature_engine without groups will provide for us, hence adding groups and avoiding index based cv should provide exactly same result

Reply:
ahhhhh!!! clever. I missed that.